packages/backend-output-storage/src/stack_metadata_output_storage_strategy.ts

```diff
@@ -36,4 +37,29 @@ export class StackMetadataBackendOutputStorageStrategy
       stackOutputs: Object.keys(backendOutputEntry.payload),
     });
   };
+
+  // TODO this should be pushed to interface as well.
+  addAppendableBackendOutputEntry = (
+    keyName: string
+  ): AppendableBackendOutputEntry => {
+    const outputs: Record<string, string> = {};
+    this.stack.addMetadata(keyName, {
+      // TODO we should have some discriminator to distinguish dynamic entry from typed one
+      version: '1',
+      stackOutputs: Lazy.list({
+        produce: () => {
+          return Object.keys(outputs);
+        },
+      }),
+    });
+    return {
+      appendOutput: (key: string, value: string) => {
+        if (key in outputs) {
+          throw new Error(`Output ${key} is already defined`);
+        }
+        outputs[key] = value;
+        new CfnOutput(this.stack, key, { value });
+      }
+    }
+  };
```

Comment:
The existing `addBackendOutputEntry` doesn't work well for `payload` that's produced in multiple places. Therefore I invented a new type of BackendOutputEntry that can be appended to.

In general I was considering two approaches:
1. To put all custom outputs under `AWS::Amplify::Custom` metadata key.
    1. This seems to simplify handling on client config side.
2. To have `AWS::Amplify::Custom::{key}` per each custom output.
    1. This avoids creating `addAppendableBackendOutputEntry`
    2. But it complicates client config generation part, i.e. having dynamic key prefix complicates zod parsing of outputs and "client config contributor" resolution to handle given entry.